File: Week6/ang4/src/app/employee-list/employee.ts
import { Department } from '../department-list/department';
export class Employee {
  constructor(
   public id: string,
   public firstname: string,
   public lastname: string,
   public email: string,
   private _department: Department
  ) {}

  public get department(): string {
    return this._department.name;
  }
  public get location(): string {
    return this._department.location;
  }
}

File: Week6/employee-service/src/main/java/com/example/employeeservice/Employee.java
package com.example.employeeservice;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "employee_table")
public class Employee {
  @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
  public int id;
  public String firstName;
  public String lastName;
  public String email;
  public String department;
}
